cmd/up.go

```diff
@@ -80,9 +86,107 @@ func newUpCmd() *cobra.Command {
 				Debug:                debug,
 			}
 
-			s, err := requireStack(stack, true, opts.Display)
-			if err != nil {
-				return err
+			var c config.Map
+			var s backend.Stack
+			hasStack := false
+
+			if len(args) > 0 {
+				url := args[0]
+
+				if !workspace.IsTemplateURL(url) {
+					return errors.Errorf("%s is not a valid URL", url)
+				}
+
+				// Retrieve the template repo.
+				var repo workspace.TemplateRepository
+				if repo, err = workspace.RetrieveTemplates(url, false); err != nil {
+					return err
+				}
+
+				// List the templates from the repo.
+				var templates []workspace.Template
+				if templates, err = repo.Templates(); err != nil {
+					return err
+				}
+
+				// Make sure only a single template is found.
+				// Alternatively, we could consider prompting to choose one instead of failing.
+				if len(templates) != 1 {
+					return errors.Errorf("more than one app template found at %s", url)
+				}
+				template := templates[0]
+
+				// Create temp directory for the "virtual workspace".
+				var temp string
+				if temp, err = ioutil.TempDir("", "pulumi-up-"); err != nil {
+					return err
+				}
+
+				// TODO don't use template name/description for project name/description.
+				// Consider prompting if they are ${PROJECT} and ${DESCRIPTION}.
+				if err = template.CopyTemplateFiles(temp, true, template.Name, template.Description); err != nil {
```

Comment:
The way we've done templates previously, the `Pulumi.yaml` will have `${PROJECT}` and `${DESCRIPTION}` specified for `name` and `description`, which get replaced by real values specified by the user (via `new`) when the template is copied over to it's destination directory. For `up` this doesn't make much sense. We don't want to prompt for these values, `up` templates should just have these values already filled in the `Pulumi.yaml`. Need to clean this up here. (Optionally, if using `up` with a template that does have `${PROJECT}` and `${DESCRIPTION}` for these, we could consider prompting in that case).